drivers/mbox/mbox_ti_omap.c

```diff
@@ -0,0 +1,196 @@
+	struct omap_mailbox_data *data = dev->data;
+	volatile struct omap_mailbox_regs *regs = (struct omap_mailbox_regs *)data->mapped_base;
+
+	if (channel >= MAILBOX_MAX_CHANNELS) {
+		return -EINVAL;
+	}
+
+	if (!msg) {
+		if (!regs->fifo_status[channel]) {
+			regs->message[channel] = 0;
+		}
+		return 0;
+	}
+
+	if (msg->size != MAILBOX_MBOX_SIZE) {
+		return -EMSGSIZE;
+	}
+
+	memcpy(&data32, msg->data, msg->size);
+
+	if (!regs->fifo_status[channel]) {
+		regs->message[channel] = data32;
+	}
+
+	return 0;
+}
+
+static int omap_mailbox_register_callback(const struct device *dev, uint32_t channel,
+					  mbox_callback_t cb, void *user_data)
+{
+	struct omap_mailbox_data *data = dev->data;
+
+	if (channel >= MAILBOX_MAX_CHANNELS) {
+		return -EINVAL;
+	}
+
+	data->cb[channel] = cb;
+	data->user_data[channel] = user_data;
+
+	return 0;
+}
+
+static int omap_mailbox_mtu_get(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return MAILBOX_MBOX_SIZE;
+}
+
+static uint32_t omap_mailbox_max_channels_get(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return MAILBOX_MAX_CHANNELS;
+}
+
+static int omap_mailbox_set_enabled(const struct device *dev, uint32_t channel, bool enable)
+{
+	struct omap_mailbox_data *data = dev->data;
+	const struct omap_mailbox_config *cfg = dev->config;
+	volatile struct omap_mailbox_regs *regs;
+	uint32_t irqstatus;
+
+	regs = (struct omap_mailbox_regs *)data->mapped_base;
+	if (channel >= MAILBOX_MAX_CHANNELS) {
+		return -EINVAL;
+	}
+
+	irqstatus = regs->irq_regs[cfg->usr_id].enable_set;
+	irqstatus |= MAILBOX_IRQ_NEWMSG(channel);
+	regs->irq_regs[cfg->usr_id].enable_set = irqstatus;
+	data->channel_enable[channel] = enable;
+
+	return 0;
+}
+
+static const struct mbox_driver_api omap_mailbox_driver_api = {
+	.send = omap_mailbox_send,
+	.register_callback = omap_mailbox_register_callback,
+	.mtu_get = omap_mailbox_mtu_get,
+	.max_channels_get = omap_mailbox_max_channels_get,
+	.set_enabled = omap_mailbox_set_enabled,
+};
+
+#define MAILBOX_INSTANCE_DEFINE(idx)	\
+	static struct omap_mailbox_data omap_mailbox_##idx##_data;	\
+	const static struct omap_mailbox_config omap_mailbox_##idx##_config = {	\
+		.usr_id =  DT_INST_PROP(idx, usr_id),	\
+	};	\
+	static int omap_mailbox_##idx##_init(const struct device *dev)	\
+	{	\
+		struct omap_mailbox_data *data = dev->data;	\
+		mem_addr_t addr;	\
+		device_map((mm_reg_t *)&addr, DT_INST_REG_ADDR(idx),	\
+		DT_INST_REG_SIZE(idx), K_MEM_CACHE_NONE);	\
+		data->mapped_base = (void *)addr;	\
+		IRQ_CONNECT(DT_INST_IRQN(idx), DT_INST_IRQ(idx, priority), omap_mailbox_isr,	\
+			    DEVICE_DT_INST_GET(idx), 0);	\
+		irq_enable(DT_INST_IRQN(idx));	\
+		return 0;	\
+	}	 \
+	DEVICE_DT_INST_DEFINE(idx, omap_mailbox_##idx##_init, \
+				  NULL, &omap_mailbox_##idx##_data,	\
+			      &omap_mailbox_##idx##_config, POST_KERNEL, \
+				  CONFIG_MBOX_INIT_PRIORITY,	\
+			      &omap_mailbox_driver_api)
+
+#define MAILBOX_INST(idx) MAILBOX_INSTANCE_DEFINE(idx);
```

Comment:
Looks like you could drop this and use `MAILBOX_INSTANCE_DEFINE` directly.

Reply:
Done